src/auth/AuthApiClient.ts

```diff
@@ -71,22 +72,29 @@ export class AuthApiClient {
     });
   }
 
-  async uploadUserProfileImage(image: any, token: string) {
+  async uploadUserProfileImage(image: UploadableFile, token: string) {
     const response = await fetch(`${this.endpoint}/auth/profile/image`, {
       method: "POST",
       headers: {
         Authorization: token,
+        "Content-Type": image.type,
+        "Content-Length": image.size.toString(),
       },
+      body: image.buffer.buffer,
     });
 
-    const data = await response.json();
+    if (!response.ok) {
+      let errorMessage = `Upload failed: ${response.status} ${response.statusText}`;
```

Comment:
I think just throwing a custom error instead of building an error message string is better here.